packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx

```diff
@@ -205,11 +216,38 @@ const MetricsVisualization: FC<Props> = ({
         };
     }, [activeData]);
 
+    const segmentedData = useMemo((): {
```

Comment:
can we do this processing on the BE instead?

Reply:
its not a good idea because segmented data derives from the date zoom. I actually had it that way initially but it adds more code